### ISSUE TITLE
UI Graph on inverts colors/sections on refresh randomly

### ISSUE THREAD

**Issue by japerry911** [kestra-io/kestra#6603]
### Describe the issue

- UI Graph on inverts colors/sections on refresh randomly

- as I click Refresh, the graph sometimes changes, see video for example

https://www.loom.com/share/041b04c1f937482ca33824f1cba26486?sid=e5389c5c-2931-40b2-bd5a-14ea406778df

Thank you, let me know if you need any other details.

### Environment

- Kestra Version: v0.20.7

**Comment by Ben8t** [comment]
Indeed, seems a bit odd. @MilosPaunovic do you know if ordering is set by defaults? I would expect to have bars ordered from bottom to top by order of amount.

**Comment by MilosPaunovic** [comment]
It's currently  not being ordered on the `UI` side, it's shown only what the `BE` returns as a response, and there it's a matter of how `DB` parses it.

I've assigned myself here to add the ordering on the `UI` side, like you've said, by amount values.

**Comment by MilosPaunovic** [comment]
I am having trouble reproducing this on latest `develop`. There was some work recently on the code related to it, so it might've been amended by it. Could you check it again @japerry911 to confirm is it still an issue?

**Comment by japerry911** [comment]
Hello @MilosPaunovic !

It is still occurring when I refresh with refresh button, every few refreshes it inverses the stack column charts. Let me know if you need a Loom, can record one before tomorrow if so.

Kestra Version - v0.20.12

**Comment by MilosPaunovic** [comment]
Hey, sorry for not being to clear about it - I was trying to reproduce it on the code which turned out to be released as `0.21.0` yesterday, but if you've still experiencing the problem on `0.20.x`, then I'll move my focus there. Thanks for the quick response! 🚀

**Comment by MilosPaunovic** [comment]
I am having real trouble trying to reproduce this.

Tried on exact version `0.20.7`, tried on latest minor one of 20, `0.20.15` and tried on current `develop` branch, no luck.

Added fake server response to simulate different states, see video, still no luck in reproduction.
https://github.com/user-attachments/assets/137b058f-10af-45c1-a55e-3974493f8e28

Could one of you give me a hand here @Ben8t or @smunteankestra?

**Comment by japerry911** [comment]
Hey @MilosPaunovic , this is still occurring in 0.21.3. Lemme know if you want a quick loom or anything.

**Comment by MilosPaunovic** [comment]
Hi Jack,

Thanks for the offer, but we'd need to reproduce this on the code side somehow. I have an idea that might help; I'll try it in a couple of days, once I clear up some of the other stuff.

**Comment by MilosPaunovic** [comment]
Okay, I'm confident that I've found what the cause is and that it's now solved on `develop`, plus, the change is backported to two previous versions. It will be available on the next maintenance release, the following Tuesday, as part of `v0.21.6` and `v0.20.19`.

I'll close this issue for now, if there is a need to re-open it after the deployment and testing, feel free to do it @japerry911.

**Comment by japerry911** [comment]
Hey @MilosPaunovic , thank you x 1000 for looking into this and helping build great product, sorry for delay on this. I would like to reopen if possible as it's still happening in v0.21.9. This is not urgent though, it's sort of lowest priority, just a UI bug I noticed.

Let me know your thoughts and/or if it reproduces.

Here's loom showing it (first tenant is hard to tell, second is easier)

https://www.loom.com/share/eb596a8d884a4eeb826b0c14177a3711?sid=b6335979-a6cb-4f82-b2e4-88c91d7fa6fb

**Comment by MilosPaunovic** [comment]
Hi @japerry911, yeah, no worries, re-opening it and putting back to backlog, to be looked at.
